control-plane-operator/controllers/hostedcontrolplane/olm/catalogs.go

```diff
@@ -118,19 +128,53 @@ func findTagReference(tags []imagev1.TagReference, name string) *imagev1.TagRefe
 	return nil
 }
 
-var CatalogToImage = map[string]string{
-	"certified-operators": "registry.redhat.io/redhat/certified-operator-index:v4.16",
-	"community-operators": "registry.redhat.io/redhat/community-operator-index:v4.16",
-	"redhat-marketplace":  "registry.redhat.io/redhat/redhat-marketplace-index:v4.16",
-	"redhat-operators":    "registry.redhat.io/redhat/redhat-operator-index:v4.16",
+func GetCatalogImages(ctx context.Context, hcp hyperv1.HostedControlPlane, pullSecret []byte) (map[string]string, error) {
+	imageRef := hcp.Spec.ReleaseImage
+	imageConfig, _, _, err := registryclient.GetMetadata(ctx, imageRef, pullSecret)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get image metadata: %w", err)
+	}
+
+	version := strings.Split(imageConfig.Config.Labels["io.openshift.release"], ".")
```

Comment:
we should just parse this into a `semver.Version` and manipulate that
```suggestion
	version := semver.Parse(imageConfig.Config.Labels["io.openshift.release"])	
```